gst/tensor_transform/CMakeLists.txt

```diff
@@ -1 +1,22 @@
-ADD_LIBRARY(tensor_transformOBJ OBJECT tensor_transform.c)
+SET(TRANSFORM_SRCS
+	tensor_transform.c
+)
+
+IF(DEFINED HAVE_ORC AND HAVE_ORC)
+	SET(ORC_SOURCE transform-orc)
+	SET(ORC_OUT_PREFIX ${CMAKE_CURRENT_SOURCE_DIR}/${ORC_SOURCE})
+
+	# Generate orc .h and .c files
+	ADD_CUSTOM_COMMAND(
+		OUTPUT ${ORC_OUT_PREFIX}.h ${ORC_OUT_PREFIX}.c
+		COMMAND rm -f ${ORC_OUT_PREFIX}.h
+		COMMAND rm -f ${ORC_OUT_PREFIX}.c
+		COMMAND orcc --implementation --include glib.h -o ${ORC_OUT_PREFIX}.c ${ORC_OUT_PREFIX}.orc
+		COMMAND orcc --header --include glib.h -o ${ORC_OUT_PREFIX}.h ${ORC_OUT_PREFIX}.orc
+		DEPENDS ${ORC_OUT_PREFIX}.orc
+	)
```

Comment:
Don't you have the issue of dependencies between ORC_OUTPUT and transformOBJ ?

Reply:
When changed orc file, it regenerates c/h files and check the dependency of tensor_transformOBJ.

```
[ 17%] Generating ../../../gst/tensor_transform/transform-orc.h, ../../../gst/tensor_transform/transform-orc.c
Scanning dependencies of target tensor_transformOBJ
[ 18%] Building C object gst/tensor_transform/CMakeFiles/tensor_transformOBJ.dir/tensor_transform.c.o
[ 19%] Building C object gst/tensor_transform/CMakeFiles/tensor_transformOBJ.dir/transform-orc.c.o
```

Reply:
The execution order in CMake is not guaranteed. Without explicit dependencies there are possibilities that tensor_transofrmOBJ is built before tranform-orc.

Due to the non-deterministic behaviors without explicit dependencies, with ```maje -j8```, you may see:
```
...
[ 18%] Building C object gst/tensor_transform/CMakeFiles/tensor_transformOBJ.dir/tensor_transform.c.o
...
[ 20%] Building C object gst/tensor_transform/CMakeFiles/tensor_transformOBJ.dir/transform-orc.c.o
...
[ 28%] Built target tensor_transformOBJ
```
which is fine with today's tensor_transform.c, but will break the build with tensor_transform.c using transform-orc.h next week.

Reply:
Yes, it will generate transform-orc.h/c. Can you **guarantee** that they will be generated before compiling tensor_transform.c ?

Reply:
add dependency for tensor_transform.c tensor_transform.h
DEPENDS ${ORC_SOURCE}.orc tensor_transform.c tensor_transform.h